### PR TITLE
[VR-10902] Async log environment in run

### PR DESCRIPTION
**Pull request by anandjakhaniya**
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-10902

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 1]
```diff
@@ -874,4 +877,30 @@ public FutureExperimentRunDAO(
...
+                    handle ->
+                        handle
+                            .createUpdate(
+                                "UPDATE experiment_run SET environment = :environment WHERE id = :runId")
+                            .bind("runId", request.getId())
+                            .bind(
+                                "environment",
+                                environmentHandler.getEnvironmentStringFromBlob(
+                                    request.getEnvironment()))
+                            .execute()),
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Why is this not in the environment handler?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java [hunk 2]
```diff
@@ -232,10 +229,7 @@ private ExperimentRun getExperimentRunFromRequest(
               runValueMap.put("parent_id", newExperimentRun.getParentId());
               runValueMap.put("owner", newExperimentRun.getOwner());
 
-              EnvironmentBlob environmentBlob =
-                  sortPythonEnvironmentBlob(newExperimentRun.getEnvironment());
-              runValueMap.put(
-                  "environment", ModelDBUtils.getStringFromProtoObject(environmentBlob));
+              runValueMap.put("environment", null);
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: Seems like we're just saving the wrong value now. Why this change?

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: - Before this async DAO we are not set any value in database at creation time so in database that column has NULL value for each new runs.
- but then we have introduce new DAO and i forgot that logic so i have implemented `ModelDBUtils.getStringFromProtoObject(environmentBlob)` instead of null so it will create empty JSON in database while creating each runs
- So Now while implementing `logEnvironment` endpoint i realize that that logic is not right so i have added null here. it will not affect any user or  code.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: if you would mind to take empty JSON like `{}` database instead of null then this explicitly NULL set not needed.

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Moved to environment handler

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java (hunk 2 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: I don't mind either way, as long as we keep the behavior.

**Review by conradoverta — APPROVED** [review]
